tasks/main.yml

```diff
@@ -1,35 +1,97 @@
+# roles/keystone/tasks/main.yml
 ---
-# tasks file for ansible-role-openio-keystone
-- name: 'Fail if required variables are not set'
-  fail:
-    msg: "{{ item }} variable must be set to apply this role."
-  when: (item is undefined) or (item is none)
-  with_items: "{{ openio_keystone_required_vars }}"
-
 - name: "Include {{ ansible_distribution }} variables"
   include_vars: "{{ item }}"
   with_first_found:
     - "{{ ansible_distribution }}.yml"
     - "{{ ansible_os_family }}.yml"
+  tags: install
+
+- name: "Include {{ ansible_distribution }} tasks"
+  include_tasks: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+  tags: install
+
+- name: Test directories presence
+  stat:
+    path: "{{ item }}"
+  register: _dir
+  with_items:
+    - "/etc/gridinit.d/{{ openio_keystone_namespace }}"
+    - "{{ openio_keystone_sysconfig_dir }}/keystone-{{ openio_keystone_serviceid }}"
 
-- include: keystone_install.yml
-  static: no
+- name: Ensure directories exists
+  file:
+    path: "{{ item.item }}"
+    state: directory
+    owner: openio
+    group: openio
+    mode: 0755
+    recurse: true
+  with_items: "{{ _dir.results }}"
+  when: not item.stat.exists
+  tags: install
+- name: 'Install configuration files 1/2'
+  copy:
+    src: keystone-paste.ini
+    dest: "{{ openio_keystone_config_dir }}/keystone-paste.ini"
+    owner: root
+    group: "{{ openio_keystone_system_group_name }}"
+    mode: '0640'
+  tags: configure
 
-- include: keystone_db_setup.yml
+- name: 'Install configuration files 2/2'
+  template:
+    src: keystone.conf.j2
+    dest: "{{ openio_keystone_config_dir }}/keystone.conf"
+    owner: root
+    group: "{{ openio_keystone_system_group_name }}"
+    mode: '0640'
+  tags: configure
 
-- include: keystone_fernet.yml
-  static: no
+- import_tasks: db_setup.yml
+  tags: configure
+
+- import_tasks: fernet.yml
   when:
     - "'fernet' in openio_keystone_token_provider"
 
-- include: keystone_credentials.yml
-  static: no
+- import_tasks: credentials.yml
+
+- name: Generate configuration files (uwsgi)
+  template:
+    src: "keystone-uwsgi.ini.j2"
+    dest: "{{ openio_keystone_sysconfig_dir }}/keystone-{{ openio_keystone_serviceid }}/{{ item }}.ini"
+    mode: "0644"
+  with_items: "{{ openio_keystone_wsgi_program_names }}"
+  register: _uwsgi_conf
+
+- name: Generate configuration files (gridinit)
+  template:
+    src: "gridinit_keystone.conf.j2"
+    dest: "{{ openio_keystone_gridinit_dir }}/{{ openio_keystone_gridinit_file_prefix }}\
+      keystone-{{ openio_keystone_serviceid }}.{{ item.0 }}.conf"
+    mode: "0644"
+  with_indexed_items: "{{ openio_keystone_wsgi_program_names }}"
+  register: _gridinit_conf
 
-- include: keystone_uwsgi.yml
-  static: no
+- name: restart keystone
```

Comment:
Use gridinit role's python library module, or at least separate the 3 commands, in case one of them fail, we want to know which one